.travis.yml

```diff
@@ -1,10 +1,83 @@
-language: node_js
-node_js: stable
+_aliases:
+  - &node
+    addons:
+      firefox: latest-beta
+    language: node_js
+    node_js: stable
+    before_install: skip
+    before_script: skip
 
-addons:
-  firefox: latest-beta
+jobs:
+  include:
+    - stage:
+      language: python
+      python: 3.6
+      addons:
+        firefox: latest-nightly
+      apt:
+        packages:
+          - jwm
+      env: TOXENV=integration-tests MOZ_HEADLESS=1 GECKODRIVER=0.19.0
+      install: 
+        - nvm install node
+        - pip install tox-travis
+      before_script:
+        - npm install
+        - npm run package
+      script: tox
+      after_success: skip
+      before_deploy: skip
+    - stage:
+      language: python
+      python: 3.6
+      env: TOXENV=flake8
+      before_install: skip
+      install:
+        - pip install tox-travis
+      before_script: skip
+      script: tox
+      after_success: skip
+      before_deploy: skip
+    - stage:
+      <<: *node
+      before_deploy: skip
+    - stage: deploy
+      if: branch = master
+      <<: *node
+      script: skip
+      deploy:
+        provider: pages
+        on:
+          branch: master
+        skip_cleanup: true
+        local_dir: site
+        github_token: $GITHUB_TOKEN
+# deploy tags to GitHub Releases
+    - stage: deploy
+      if: tag =~ ^\d.\d.\d
+      <<: *node
+      script: skip
+      deploy:
+        provider: releases
+        on:
+          tags: true
+        skip_cleanup: true
+        file_glob: true
+        file: addon.xpi
+        api_key: $GITHUB_TOKEN
+
+before_install:
+  - wget -O /tmp/geckodriver.tar.gz https://github.com/mozilla/geckodriver/releases/download/v$GECKODRIVER/geckodriver-v$GECKODRIVER-linux64.tar.gz
+  - mkdir $HOME/geckodriver && tar xvf /tmp/geckodriver.tar.gz -C $HOME/geckodriver
+  - export PATH=$HOME/geckodriver:$PATH
+  - firefox --version
+  - geckodriver --version
 
-# DEFAULT: Travis runs `npm install`
+before_script:
+  - sh -e /etc/init.d/xvfb start
+  - sleep 10
+  - jwm &
+  - sleep 10
```

Comment:
If we're using MOZ_HEADLESS, do we need this?

Reply:
It seems to help, without it, I get a lot of timeouts

Reply:
Hmm.. we don't use these in FoxPuppet for Firefox beta and later, and they're fine?

Reply:
Maybe because this is more of a context test whereas foxpuppet is chrome. 

I was getting failures like this just using xvfb: https://travis-ci.org/jrbenny35/lockbox-extension/jobs/295418535 and sometimes other random errors.

Reply:
Ah, maybe the multiple tabs needs JWM.. I think that makes sense. You're not using xvfb though, because you're not setting a DISPLAY environment variable. I'm pretty sure it's not needed.

Reply:
Let's remove xvfb as it's not needed for running in headless mode.